src/nvim/diff.c

```diff
@@ -1116,68 +1119,50 @@ void diff_win_options(win_T *wp, int addbuf)
 /// @param eap
 void ex_diffoff(exarg_T *eap)
 {
-  win_T *old_curwin = curwin;
-  int diffwin = FALSE;
+  int diffwin = false;
 
   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
     if (eap->forceit ? wp->w_p_diff : (wp == curwin)) {
-      // Set 'diff', 'scrollbind' off and 'wrap' on. If option values
-      // were saved in diff_win_options() restore them.
-      wp->w_p_diff = FALSE;
-
-      if (wp->w_p_scb) {
-        wp->w_p_scb = wp->w_p_diff_saved ? wp->w_p_scb_save : FALSE;
-      }
-
-      if (wp->w_p_crb) {
-        wp->w_p_crb = wp->w_p_diff_saved ? wp->w_p_crb_save : FALSE;
-      }
-
-      if (!wp->w_p_wrap) {
-        wp->w_p_wrap = wp->w_p_diff_saved ? wp->w_p_wrap_save : TRUE;
-      }
-      curwin = wp;
-      curbuf = curwin->w_buffer;
+      // Set 'diff' off. If option values were saved in
+      // diff_win_options(), restore the ones whose settings seem to have
+      // been left over from diff mode.
+      wp->w_p_diff = false;
 
       if (wp->w_p_diff_saved) {
-        free_string_option(wp->w_p_fdm);
-        wp->w_p_fdm = wp->w_p_fdm_save;
-        wp->w_p_fdm_save = empty_option;
-      } else {
-        set_string_option_direct((char_u *)"fdm", -1,
-                                 (char_u *)"manual", OPT_LOCAL | OPT_FREE, 0);
-      }
-      curwin = old_curwin;
-      curbuf = curwin->w_buffer;
+        if (wp->w_p_scb) {
+          wp->w_p_scb = wp->w_p_diff_saved ? wp->w_p_scb_save : false;
+        }
 
-      if (wp->w_p_fdc == diff_foldcolumn) {
-        wp->w_p_fdc = wp->w_p_diff_saved ? wp->w_p_fdc_save : 0;
-      }
+        if (wp->w_p_crb) {
+          wp->w_p_crb = wp->w_p_diff_saved ? wp->w_p_crb_save : false;
+        }
 
-      if ((wp->w_p_fdl == 0)
-          && wp->w_p_diff_saved) {
-        wp->w_p_fdl = wp->w_p_fdl_save;
-      }
+        if (!wp->w_p_wrap) {
+          wp->w_p_wrap = wp->w_p_diff_saved ? wp->w_p_wrap_save : true;
+        }
```

Comment:
For this trio of conditions (starting from line 1132), patch at https://github.com/vim/vim/commit/4392996461f56803d5c4a4919ba6dc509434bef5 removed the ternary, so it looks like this:

    wp->w_p_scb = wp->w_p_scb_save;